src/test/regress/expected/multi_transaction_recovery.out

```diff
@@ -331,6 +331,104 @@ SELECT count(*) FROM pg_dist_transaction;
      2
 (1 row)
 
+-- check that read-only participants skip prepare
+-- only two of the connections will perform a write (INSERT)
+SET citus.shard_count TO 4;
+CREATE TABLE test_2pcskip (a int);
+SELECT create_distributed_table('test_2pcskip', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO test_2pcskip SELECT i FROM generate_series(0, 5)i;
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions
+---------------------------------------------------------------------
+                             0
+(1 row)
+
+BEGIN;
+-- these inserts use two connections
```

Comment:
This information is vague. Can we do something like to ensure that these values goes to different shards on different workers?:

```
SELECT count(DISTINCT nodeport) FROM pg_dist_shard_placement WHERE shardid IN (get_shard_id_for_distribution_column('test', 6),get_shard_id_for_distribution_column('test', 7));
``` 

Otherwise if shard count/worker count changes, this test may not work as intended

Reply:
Good idea

Reply:
> SELECT count(DISTINCT nodeport) FROM pg_dist_shard_placement WHERE shardid IN (get_shard_id_for_distribution_column('test', 6),get_shard_id_for_distribution_column('test', 7));

test should be `test_2pcskip`, that is the table name if you are not familiar with this UDF.